cluster/provision_gateway.sh

```diff
@@ -0,0 +1,675 @@
+      sed -i 's/# control_network_ip=/control_network_ip='$MINION_OVERLAY_NET_IP'/g' $vrac
+      sed -i 's/# name=vhost0/name=vhost0/g' $vrac
+      sed -i 's,# ip=10.1.1.1/24,ip='$MINION_OVERLAY_NET_IP/$cidr',g' $vrac
+      if [ "$def" == "default" ]; then
+         defgw=$(ip route | grep $OPENCONTRAIL_VROUTER_INTF | grep $def | awk 'NR==1{print $3}')
+      else
+         defgw=$(ip route | grep $OPENCONTRAIL_VROUTER_INTF | grep '$via' | awk 'NR==1{print $3}')
+      fi
+      if [ -z $defgw ]; then
+         # assuming .1 is the gw for the range which is what kubernetes network provider 
+         # assigns. If found different we need to change it
+         defgw=$(sipcalc $OPENCONTRAIL_VROUTER_INTF | grep '$ur' | awk '{print $4}')
+      fi
+      if [ -z $defgw ]; then
+         # check IP on VHOST
+         vip=$(ip a | grep $VHOST | grep inet | awk -F/ '{print $1}' | awk '{print $2}')
+         if [ $vip == $MINION_OVERLAY_NET_IP ]; then
+            defgw=$(ip route | grep $VHOST | grep $via | awk 'NR==1{print $3}')
+         fi
+      fi
+      sed -i 's,# gateway=10.1.1.254,gateway='$defgw',g' $vrac
+      if isGceVM ; then
+          sed -i 's,# subnet_hosts_resolvable=0,subnet_hosts_resolvable=0,g' $vrac
+      fi
+      sed -i 's/# physical_interface=vnet0/physical_interface='$OPENCONTRAIL_VROUTER_INTF'/g' $vrac
+      sed -i 's/compute_node_address = 10.204.216.28/# compute_node_address = /g' $vrac
+  fi
+  wget -P /tmp https://raw.githubusercontent.com/Juniper/contrail-kubernetes/$ockver/cluster/contrail-vrouter-agent.manifest
+  vragentfile=/tmp/contrail-vrouter-agent.manifest
+  vrimg=$(cat $vragentfile | grep image | awk -F, '{print $1}' | awk '{print $2}')
+  echo $vrimg | xargs -n1 sudo docker pull
+  i=0
+  while true
+    do
+      dvrimg=$(docker images | grep -ow vrouter-agent)
+      if [ ! -z  "$dvrimg" ]; then
+         break
+      fi
+      sleep 5
+      ((i++))
+      if [ $i -eq 12 ]; then
+       if [ -d "/proc/${pullpid}" ]; then
+          pkill -TERM -P $pullpid
+          cnt=$(ps -ef|grep "docker pull" | grep vrouter-agent | wc -l)
+          if [ $cnt -gt 1 ]; then
+             log_info_msg "Restarting docker and retry pull vrouter image"
+             service docker restart
+          fi
+       fi
+       # give time for docker to initialize
+       sleep 60
+       log_info_msg "pulling of opencontrail/vrouter-agent image was not successful in the initial attempt."
+       (echo $vrimg | xargs -n1 sudo docker pull) & pullpid=$!
+       i=0
+      fi
+    done
+}
+
+function vr_agent_manifest_setup()
+{
+  # Keeping the manifest as in a minion
+  # there is no kubelet on a gateway node
+  if [ ! -f /etc/kubernetes/manifests ]; then
+     mkdir -p /etc/kubernetes/manifests
+  fi
+  vra_manifest="/etc/kubernetes/manifests/contrail-vrouter-agent.manifest"
+  if [ -f "$vra_manifest" ]; then
+     rm -f "$vra_manifest"
+  fi
+  # Wait for the control node to be up
```

Comment:
The provisioning scripts should not have the nodes waiting for each other if this can be avoided.